src/bots/fillerLite.ts

```diff
@@ -0,0 +1,1274 @@
+							logger.info(
+								`parse logs took ${processBulkFillLogsDuration}ms, filled ${successfulFills} (fillTxId: ${fillTxId})`
+							);
+
+							// record successful fills
+							const user = this.driftClient.getUser();
+							this.successfulFillsCounter.add(
+								successfulFills,
+								metricAttrFromUserAccount(
+									user.userAccountPublicKey,
+									user.getUserAccount()
+								)
+							);
+						})
+						.catch((e) => {
+							// console.error(e);
+							logger.error(
+								`Failed to process fill tx logs (error above) (fillTxId: ${fillTxId}):`
+							);
+							webhookMessage(
+								`[${this.name}]: :x: error processing fill tx logs:\n${
+									e.stack ? e.stack : e.message
+								}`
+							);
+						});
+				})
+				.catch(async (e) => {
+					// console.error(e);
+					logger.error(
+						`Failed to send packed tx (error above) (fillTxId: ${fillTxId}):`
+					);
+					const simError = e as SendTransactionError;
+
+					if (simError.logs && simError.logs.length > 0) {
+						const start = Date.now();
+						await this.handleTransactionLogs(nodesSent, simError.logs);
+						logger.error(
+							`Failed to send tx, sim error tx logs took: ${
+								Date.now() - start
+							}ms (fillTxId: ${fillTxId})`
+						);
+
+						const errorCode = getErrorCode(e);
+
+						if (
+							!errorCodesToSuppress.includes(errorCode) &&
+							!(e as Error).message.includes('Transaction was not confirmed')
+						) {
+							if (errorCode) {
+								this.txSimErrorCounter.add(1, {
+									errorCode: errorCode.toString(),
+								});
+							}
+							webhookMessage(
+								`[${this.name}]: :x: error simulating tx:\n${
+									simError.logs ? simError.logs.join('\n') : ''
+								}\n${e.stack || e}`
+							);
+						}
+					}
+				})
+				.finally(() => {
+					this.removeFillingNodes(nodesSent);
+				});
+		}
+	}
+
+	/**
+	 * It's difficult to estimate CU cost of multi maker ix, so we'll just send it in its own transaction
+	 * @param node node with multiple makers
+	 */
+	protected async tryFillMultiMakerPerpNodes(nodeToFill: NodeToFill) {
+		const ixs: Array<TransactionInstruction> = [];
+		const fillTxId = this.fillTxId++;
+		logger.info(
+			logMessageForNodeToFill(
+				nodeToFill,
+				`Filling multi maker perp node with ${nodeToFill.makerNodes.length} makers (fillTxId: ${fillTxId})`
+			)
+		);
+
+		try {
+			// first ix is compute budget
+			const computeBudgetIx = ComputeBudgetProgram.requestUnits({
+				units: 10_000_000,
+				additionalFee: 0,
+			});
+			ixs.push(computeBudgetIx);
+
+			const { makerInfos, takerUser, referrerInfo, marketType } =
+				await this.getNodeToFillInfo(nodeToFill);
+
+			if (!isVariant(marketType, 'perp')) {
+				throw new Error('expected perp market type');
+			}
+
+			ixs.push(
+				await this.driftClient.getFillPerpOrderIx(
+					await getUserAccountPublicKey(
+						this.driftClient.program.programId,
+						takerUser.authority
+					),
```

Comment:
dont think this PDA is correct, since it is assuming subaccountId 0 for the taker.

Reply:
oh you can get that from `takerUser.subaccountId`